server/src/main/java/org/elasticsearch/transport/RemoteClusterPortSettings.java

```diff
@@ -0,0 +1,175 @@
+import static org.elasticsearch.transport.TransportSettings.PUBLISH_HOST_PROFILE;
+import static org.elasticsearch.transport.TransportSettings.PUBLISH_PORT_PROFILE;
+import static org.elasticsearch.transport.TransportSettings.TCP_KEEP_ALIVE_PROFILE;
+import static org.elasticsearch.transport.TransportSettings.TCP_KEEP_COUNT_PROFILE;
+import static org.elasticsearch.transport.TransportSettings.TCP_KEEP_IDLE_PROFILE;
+import static org.elasticsearch.transport.TransportSettings.TCP_KEEP_INTERVAL_PROFILE;
+import static org.elasticsearch.transport.TransportSettings.TCP_NO_DELAY_PROFILE;
+import static org.elasticsearch.transport.TransportSettings.TCP_RECEIVE_BUFFER_SIZE_PROFILE;
+import static org.elasticsearch.transport.TransportSettings.TCP_REUSE_ADDRESS_PROFILE;
+import static org.elasticsearch.transport.TransportSettings.TCP_SEND_BUFFER_SIZE_PROFILE;
+
+/**
+ * Contains the settings and some associated logic for the settings related to the Remote Access port, used by Remote Cluster Security 2.0.
+ */
+public class RemoteClusterPortSettings {
+    private static final Logger logger = LogManager.getLogger(RemoteClusterPortSettings.class);
+
+    public static final String REMOTE_CLUSTER_PROFILE = "_remote_cluster";
+    public static final String REMOTE_CLUSTER_PREFIX = "remote_cluster.";
+    public static final String REMOTE_CLUSTER_SSL_PREFIX = REMOTE_CLUSTER_PREFIX + "ssl.";
+
+    public static final Setting<Boolean> REMOTE_CLUSTER_PORT_ENABLED = boolSetting(
+        REMOTE_CLUSTER_PREFIX + "enabled",
+        false,
+        Setting.Property.NodeScope
+    );
+
+    public static final Setting<List<String>> HOST = listSetting(
+        REMOTE_CLUSTER_PREFIX + "host",
+        TransportSettings.BIND_HOST,
+        Function.identity(),
+        Setting.Property.NodeScope
+    );
+    public static final Setting<List<String>> PUBLISH_HOST = listSetting(
+        REMOTE_CLUSTER_PREFIX + "publish_host",
+        HOST,
+        Function.identity(),
+        Setting.Property.NodeScope
+    );
+    public static final Setting<List<String>> BIND_HOST = listSetting(
+        REMOTE_CLUSTER_PREFIX + "bind_host",
+        HOST,
+        Function.identity(),
+        Setting.Property.NodeScope
+    );
+
+    public static final Setting<Integer> PORT = intSetting(REMOTE_CLUSTER_PREFIX + "port", 9900, 0, 65535, Setting.Property.NodeScope);
+
+    public static final Setting<Integer> PUBLISH_PORT = intSetting(
+        REMOTE_CLUSTER_PREFIX + "publish_port",
+        -1,
+        -1,
+        Setting.Property.NodeScope
+    );
+
+    public static final Setting<Boolean> TCP_KEEP_ALIVE = boolSetting(
+        REMOTE_CLUSTER_PREFIX + "tcp.keep_alive",
+        TransportSettings.TCP_KEEP_ALIVE,
+        Setting.Property.NodeScope
+    );
+
+    public static final Setting<Integer> TCP_KEEP_IDLE = intSetting(
+        REMOTE_CLUSTER_PREFIX + "tcp.keep_idle",
+        TransportSettings.TCP_KEEP_IDLE,
+        -1,
+        300,
```

Comment:
This min/max was taken from the general setting, so it's possible we would want a different range - I haven't thought through all the use cases. It might be good to use a constant, but I think I'd also want to adapt the existing TCP & profile settings to use that constant, which would inflate this PR - I'll write it down for next time we address tech debt.